### DISCUSSION TITLE
create-directus-project run failured

### DISCUSSION THREAD

**Discussion by liuyang3688** [directus/directus#13168] [Archive]
### Describe the Bug

Steps:

Error:
> gyp: Undefined variable module_name in binding.gyp while trying to load binding.gyp

Path:
1. create-directus-project.cmd 
2. npm install directus 
3. npm install argon2 
4. node-gyp compile 
5. compile need configure module_name & module_path. But I didnt found pos.

Question:
I dont know exactly where to configure variables above? 

If somebody knows and would like inform me, appreciated and thanks.

### To Reproduce

```shell
npm init directus-project sample
```

OR

```shell
yarn create directus-project sample
```

### Errors Shown

```shell
npm ERR! path E:\ruilang\qserver\node_modules\argon2
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c node-gyp rebuild
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@8.4.1
npm ERR! gyp info using node@16.14.2 | win32 | x64
npm ERR! gyp info find Python using Python version 3.8.0 found at "E:\Devenv\Python\Python38\python.exe"
npm ERR! gyp info find VS using VS2022 (17.1.32421.90) found at:
npm ERR! gyp info find VS "C:\Program Files\Microsoft Visual Studio\2022\Community"
npm ERR! gyp info find VS run with --verbose for detailed information
npm ERR! gyp info spawn E:\Devenv\Python\Python38\python.exe
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   'E:\\ruilang\\qserver\\node_modules\\node-gyp\\gyp\\gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'msvs',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   'E:\\ruilang\\qserver\\node_modules\\argon2\\build\\config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   'E:\\ruilang\\qserver\\node_modules\\node-gyp\\addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   'C:\\Users\\Administrator\\AppData\\Local\\node-gyp\\Cache\\16.14.2\\include\\node\\common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=C:\\Users\\Administrator\\AppData\\Local\\node-gyp\\Cache\\16.14.2',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=E:\\ruilang\\qserver\\node_modules\\node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=C:\\\\Users\\\\Administrator\\\\AppData\\\\Local\\\\node-gyp\\\\Cache\\\\16.14.2\\\\<(target_arch)\\\\node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=E:\\ruilang\\qserver\\node_modules\\argon2',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'E:\\ruilang\\qserver\\node_modules\\argon2\\build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp: Undefined variable module_name in binding.gyp while trying to load binding.gyp
npm ERR! gyp ERR! configure error
npm ERR! gyp ERR! stack Error: `gyp` failed with exit code: 1
npm ERR! gyp ERR! stack     at ChildProcess.onCpExit (E:\ruilang\qserver\node_modules\node-gyp\lib\configure.js:259:16)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:526:28)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
npm ERR! gyp ERR! System Windows_NT 10.0.19043
npm ERR! gyp ERR! command "E:\\Devenv\\nodejs\\node.exe" "E:\\ruilang\\qserver\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
npm ERR! gyp ERR! cwd E:\ruilang\qserver\node_modules\argon2
npm ERR! gyp ERR! node -v v16.14.2
npm ERR! gyp ERR! node-gyp -v v8.4.1
npm ERR! gyp ERR! not ok
```

### What version of Directus are you using?

9.10.0

### What version of Node.js are you using?

16.14.2

### What database are you using?

mysql

### What browser are you using?

chrome

### How are you deploying Directus?

running locally

**Comment by azrikahar** [comment]
Can you share where are you getting the steps in `Path:` from? Is this a manual installation, as it seems different from your reproduction steps? Were you experiencing an issue with argon2? 

Can you try running `npm rebuild argon2 --build-from-source` and retry again?

**Reply by liuyang3688** [reply]
PATH is the summary of my finding error steps. I tried install argon2 alone `npm install argon2`，same error.
The command  `npm rebuild argon2 --build-from-source`  can build argon2 successfully, but the command `directive init` failed again.

**Reply by Marcnuth** [reply]
same error for me

**Reply by azrikahar** [reply]
@liuyang3688 can you explain what is the `directive init` command? or did you mean `npm init directus-project <project-name>`?

@Marcnuth what OS and node version do you have? Feels like this is (somehow) more so something to do with argon2 🤔 Unfortunately I'm not able to replicate this issue locally on Win 10 + Node 16.15.0.

**Reply by liuyang3688** [reply]
> 

sorry for wrong spelling, correct is 

> directus init

**Reply by liuyang3688** [reply]
> > 
> 
> sorry for wrong spelling, correct is
> 
> > npx directus init

 i found the command in "/directus/directus)/packages/create-directus-project/lib/index.js" file

> await execa('npx', ['directus', 'init'], {
> 	cwd: rootPath,
> 	stdio: 'inherit',
> });

**Reply by azrikahar** [reply]
Not quite sure would this help, but can you try `npm i -g @mapbox/node-pre-gyp` and attempt `npm init directus-project <your-project-name>` again?

**Reply by liuyang3688** [reply]
> Not quite sure would this help, but can you try `npm i -g @mapbox/node-pre-gyp` and attempt `npm init directus-project <your-project-name>` again?
I have already given up, use nestjs directly instead of directus.
